app/controllers/host_reports_controller.rb

```diff
@@ -43,4 +65,24 @@ def resource_scope(options = {})
     options[:permission] = :view_host_reports
     super(options).my_reports
   end
+
+  def can_delete?(report)
+    authorized_for(auth_object: report, authorizer: authorizer, permission: "destroy_#{controller_permission}")
+  end
+
+  def can_view?(report)
+    authorized_for(auth_object: report, authorizer: authorizer, permission: "view_#{controller_permission}")
+  end
+
+  def report_status(report)
+    status = 'empty'
+    if report.failure.positive?
+      status = 'failure'
+    elsif report.change.positive?
+      status = 'change'
+    elsif report.nochange.positive?
+      status = 'nochange'
+    end
+    status
+  end
```

Comment:
I would move this into model.

I would prefer symbols over strings.

Order matters, process in this order: nochange, change, failure.

Any column can be technically `nil` so use safe navigation.

Finally no need to introduce any variable just single if and else statement should do it.

Reply:
Thanks, moved it to the model now
I need them to be specific strings I believe so in the UI I could display the correct status: https://github.com/theforeman/foreman_host_reports/pull/21/files#diff-6afe496a28034c6216c7b2cf4d553ede6c46fd4e396558015070ae07bd931428R53

as for the order, shouldn't it be the same order as in https://github.com/theforeman/foreman_host_reports/blob/develop/app/models/host_status/host_report_status.rb#L120 ?